lib/Conversion/TorchOnnxToTorch/OnnxGruExpander.cpp

```diff
@@ -0,0 +1,431 @@
+
+  int64_t hDtypeInt =
+      static_cast<int64_t>(getScalarTypeForType(hTy.getDtype()));
+  Value hDtypeIntVal = b.create<ConstantIntOp>(b.getI64IntegerAttr(hDtypeInt));
+
+  Value Y_initial = b.create<AtenZerosOp>(yTy, YShapeList, hDtypeIntVal,
+                                          cstNone, cstNone, cstNone);
+
+  Value maxTripCount = cstSeqLen;
+  Value loopConditionTrue = b.create<ConstantBoolOp>(true);
+
+  Type loopIndexType = intType;
+
+  auto loop = b.create<PrimLoopOp>(TypeRange({yTy, hTy}), maxTripCount,
+                                   loopConditionTrue,
+                                   ValueRange({Y_initial, initial_h}));
+
+  {
+    OpBuilder::InsertionGuard guard(b);
+    Block *loopBody =
+        b.createBlock(&loop.getRegion(), loop.getRegion().begin(),
+                      TypeRange({loopIndexType, yTy, hTy}), {loc, loc, loc});
+
+    Value loopIndex = loopBody->getArgument(0);
+    Value Y_prev = loopBody->getArgument(1);
+    Value H_prev = loopBody->getArgument(2);
+
+    auto XtType = b.getType<ValueTensorType>(
+        llvm::SmallVector<int64_t>{batch_size, input_size}, xTy.getDtype());
+
+    Value Xt = b.create<AtenSelectIntOp>(XtType, X, cstZero, loopIndex);
+
+    Value H_new =
+        gru_cell(b, Xt, H_prev, weights, activations, linear_before_reset);
+
+    Type hTyUnsqueezed = b.getType<ValueTensorType>(
+        llvm::SmallVector<int64_t>{1, batch_size, hidden_size}, hTy.getDtype());
+    Value H_new_unsqueezed =
+        b.create<AtenUnsqueezeOp>(hTyUnsqueezed, H_new, cstZero);
+
+    auto loopIndexPlusOne = b.create<AtenAddIntOp>(intType, loopIndex, cstOne);
+    Value Y_new =
+        b.create<AtenSliceScatterOp>(yTy, Y_prev, H_new_unsqueezed, cstZero,
+                                     loopIndex, loopIndexPlusOne, cstOne);
+
+    b.create<PrimLoopConditionOp>(loopConditionTrue,
+                                  ValueRange({Y_new, H_new}));
+  }
+
+  GruLayerOutput output;
+  output.Y = loop.getResult(0);
+  output.Y_h = loop.getResult(1);
+
+  return output;
+}
+
+LogicalResult OnnxGruExpander(OpBinder binder,
+                              ConversionPatternRewriter &rewriter) {
+  Location loc = binder.getLoc();
+  mlir::ImplicitLocOpBuilder b(loc, rewriter);
+
+  // Binding arguments
+  ValueTensorType yTy, Y_hType;
+  if (binder.tensorResultTypeAtIndex(yTy, 0) ||
+      binder.tensorResultTypeAtIndex(Y_hType, 1)) {
+    return rewriter.notifyMatchFailure(binder.op,
+                                       "At least one output must be present");
+  }
+
+  Value X, W, R, B, initial_h, sequence_lens;
+  if (binder.tensorOperandAtIndex(X, 0) || binder.tensorOperandAtIndex(W, 1) ||
+      binder.tensorOperandAtIndex(R, 2))
+    return rewriter.notifyMatchFailure(binder.op,
+                                       "Missing required input tensor");
+
+  if (binder.tensorOperandAtIndex(B, 3))
+    B = b.create<AtenZerosOp>(W.getType(), W);
```

Comment:
This needs a fix similar to the RNN expander. From the docs:

"The bias tensor for the gates. Concatenation of `[Wb[zrh], Rb[zrh]]` and `[WBb[zrh], RBb[zrh]]` (if bidirectional) along dimension 0. This tensor has shape `[num_directions, 6*hidden_size]`. "

Reply:
Fixed!